staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -101,15 +101,24 @@ func (q *queue) Dequeue() (*request, bool) {
 	return request, ok
 }
 
-// GetVirtualFinish returns the expected virtual finish time of the request at
-// index J in the queue with estimated finish time G
-func (q *queue) GetVirtualFinish(J int, G float64) float64 {
-	// The virtual finish time of request number J in the queue
-	// (counting from J=1 for the head) is J * G + (virtual start time).
-
-	// counting from J=1 for the head (eg: queue.requests[0] -> J=1) - J+1
-	jg := float64(J+1) * float64(G)
-	return jg + q.virtualStart
+// GetNextFinish returns the expected virtual finish time of the
+// oldest request in the queue with estimated finish time G
+func (q *queue) GetNextFinish(G float64) float64 {
+	var oldestReq *request
+	q.requests.Walk(func(r *request) bool {
```

Comment:
Can you add an explicit comment that if we decide to generalize this function to return time for further requests, we need to carefully evaluate and potentially improve the performance here?